gdal/doc/source/development/rfc/rfc78_osgeo_utils.rst

```diff
@@ -0,0 +1,184 @@
+.. _rfc-78:
+
+===========================================
+RFC 78: `osgeo_utils` Python package split
+===========================================
+
+============== ============================
+Author:        Idan Miara
+Contact:       idan@miara.com
+Started:       2020-Dec-3
+Last updated:  2020-Dec-3
+Status:        Preliminary, target GDAL 3.3
+============== ============================
+
+Summary
+-------
+
+This RFC is focused on the Python code in the GDAL repo, it would have no impact on non-Python code.
+This RFC suggests that the `osgeo` package will include only the SWIG bindings (As was in GDAL<3.2),
+And mainly suggests to move `osgeo.utils`, `osgeo.auxiliary` and other Python stuff that can be reused
+into a new pure-Python package named `osgeo_utils` and sets a new layout for this package.
+The `osgeo_utils` package will be distributed in pypi.org independently of the `osgeo` package.
+
+This RFC does NOT suggest exporting any code into a different repo.
+But suggests that considering moving `osgeo_utils` into an another repo could be suggested in another RFC.
+
+New Package Name
+-----------------
+This RFC sets the new package `osgeo_utils`, as the name implies, it is related to `osgeo.utils`.
+Only one character is different between `osgeo_utils` and `osgeo.utils` (which it mainly replaces)
+should help a smooth transition.
+If you think you have a better name for this package name or other any other name suggested in this RFC - please do say.
+
+Motivation
+----------
+
+As the Python code evolves independently of the GDAL core and is not directly dependent on a specific
+GDAL version, one might want use the latest `osgeo_utils` package with an older `osgeo` GDAL core, or vice versa.
```

Comment:
> As the Python code evolves independently of the GDAL core and is not directly dependent on a specific
GDAL version

Unfortunately, this is not always true. Some GDAL python scripts might rely on new GDAL C API available in updated SWIG Python bindings, or change in behaviour of existing API